pkgs/development/libraries/nlohmann_json/default.nix

```diff
@@ -1,41 +1,47 @@
-{ lib, stdenv, fetchFromGitHub, fetchpatch, cmake
+{ stdenv
+, lib
+, fetchFromGitHub
+, callPackage
+, cmake
 }:
 
 stdenv.mkDerivation rec {
   pname = "nlohmann_json";
-  version = "3.9.1";
+  version = "3.10.2";
 
   src = fetchFromGitHub {
     owner = "nlohmann";
     repo = "json";
     rev = "v${version}";
-    sha256 = "sha256-THordDPdH2qwk6lFTgeFmkl7iDuA/7YH71PTUe6vJCs=";
+    sha256 = "/OFNfukrIyfJmD0ko174aud9T6ZOesHANJjyfk4q/Vs=";
   };
 
-  patches = [
-    # https://github.com/nlohmann/json/pull/2690
-    (fetchpatch {
-      url = "https://github.com/nlohmann/json/commit/53a9850eebb88c6ff95f6042d08d5c0cc9d18097.patch";
-      sha256 = "k+Og00nXNg5IsFQY5fWD3xVQQXUFFTie44UXole0S1M=";
-    })
-  ];
-
   nativeBuildInputs = [ cmake ];
 
   cmakeFlags = [
     "-DBuildTests=${if doCheck then "ON" else "OFF"}"
     "-DJSON_MultipleHeaders=ON"
-  ];
+  ] ++ lib.lists.optional doCheck "-DJSON_TestDataDirectory=${
+    callPackage ./test_data.nix { }
+  }";
+
+  doCheck = stdenv.hostPlatform == stdenv.buildPlatform;
 
-  # A test cause the build to timeout https://github.com/nlohmann/json/issues/1816
-  #doCheck = stdenv.hostPlatform == stdenv.buildPlatform;
-  doCheck = false;
+  # The tset `cmake_fetch_content_configure` will fail if the test data is downloaded rather than fetching with git
+  # Use ctest -LE git_required to disable test `git_required` which contains `cmake_fetch_content_configure`
+  # See https://github.com/nlohmann/json/issues/2845
+  checkPhase = ''
```

Comment:
You should be able to reuse the normal checkPhase: <https://github.com/NixOS/nixpkgs/pull/134801/commits/c611fa4a51fcfde630b5adc594795ff4773986a3#diff-b08d5600eaf919f3e576d221f53ea4dcd2af1ee7af7f25605905e29dc26230e1R33>